pyaerocom/aeroval/obsentry.py

```diff
@@ -101,22 +197,20 @@ def check_add_obs(self):
                 )
             if self.obs_id not in const.OBS_IDS_UNGRIDDED:
                 try:
-                    const.add_ungridded_post_dataset(**self)
+                    const.add_ungridded_post_dataset(**self.model_dump())
                 except Exception:
                     raise InitialisationError(
                         f"Cannot initialise auxiliary reading setup for {self.obs_id}. "
                         f"Reason:\n{format_exc()}"
                     )
 
-    def get_all_vars(self) -> list:
+    def get_all_vars(self) -> tuple[str, ...]:
         """
         Get list of all variables associated with this entry
 
         Returns
         -------
-        list
-            DESCRIPTION.
-
+        tuple[str, ...]
         """
         return self.obs_vars
```

Comment:
unnecessary change of signature
```
return list(self.obs_vars)
```
would have worked as well, and didn't require the change to `(v,)` tuple code. A tuple[...] is in most cases better a list from a user-point perspective.

Reply:
This is part of an effort to make objects in the code immutable which should not change at run time, like has been changed elsewhere.

Reply:
The intention is to make obsentry immutable, which my example would also satisfy.

Reply:
I agree making `ObsEntry` immutable is great. Now implemented.

Reply:
Sadly, not as easy as I thought, but worth coming back to. #1367